### ISSUE TITLE
Exception in .Net core application reading from shapefile

### ISSUE THREAD

**Issue by FarbrorMartin** [NetTopologySuite/NetTopologySuite.IO.ShapeFile#11]
Hi!
I am trying to convert a .net standard applikation to .net core and get a runtime exception from the IO.ShapeFile library. I have added the nuget NetTopologySuite.IO.Shapefile in VS code, and my target framework is netcoreapp2.0. VS Code says everything is compatible.

The exception is thrown when reading a shapefile:
```
  GeometryFactory factory = new GeometryFactory(new PrecisionModel(PrecisionModels.Fixed),3006);
            
            var features = new List<Feature>();
            using ( ShapefileDataReader shapeFileDataReader = new ShapefileDataReader(shpFilename, factory))  {

```
It seems like maybe the DbaseEncodeUtility tries to load .Net 4.1 assemblies for System.Text.Encoding?

```
{System.IO.FileNotFoundException: Could not load file or assembly 'System.Text.Encoding.CodePages, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Text.Encoding.CodePages, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at NetTopologySuite.IO.DbaseEncodingUtility.GetEncodingForCodePageName(String name)
   at NetTopologySuite.IO.DbaseEncodingUtility..cctor() in D:\Development\Source\Repos\NTS.IO\NetTopologySuite.IO.ShapeFile\NetTopologySuite.IO.GeoTools\Dbase\DbaseEncodingUtility.cs:line 14
```

Or am I doing something wrong?

Thanks

**Comment by airbreather** [comment]
Ahh, this looks like a bug in the .nuspec file.  It's supposed to include something per [this block](https://github.com/NetTopologySuite/NetTopologySuite.IO.ShapeFile/blob/dd9a6f2c31de35ed4000995ddb26e18323bfb851/NetTopologySuite.IO.GeoTools/NetTopologySuite.IO.GeoTools.csproj#L32-L34) in the .csproj file, but it doesn't.

If I'm right, then the workaround (for now) is for you to explicitly add that same `PackageReference` to your own project and then try again.

Sorry about that, I'll get it fixed as soon as I get home from work today.

**Comment by airbreather** [comment]
See also NetTopologySuite/NetTopologySuite#244 for something that would make it harder to introduce this kind of bug, heh...

**Comment by xivk** [comment]
I created this a while back because I was stuck on this (before all the work done on NTS for .NET core support):

https://github.com/itinero/nettopologysuite.io.portableshape

Not sure if it will work for you...

**Comment by FarbrorMartin** [comment]
I think I found the problem. If I add a package reference to System.Text.Encoding.CodePage everything works fine. So maybe that package should be added as a dependency to ShapeFile?

Edit:
And after actually reading your reply I see that is exactly what you said.
Thanks for responding so quickly!

**Comment by FarbrorMartin** [comment]
I assume that Nuget should sort that dependency out automatically if it was stated in the ShapeFile package, right?

**Comment by airbreather** [comment]
Should be fixed in package version 1.15.2.
